exporter/src/test/scala/uk/gov/nationalarchives/consignmentexport/GraphQlApiSpec.scala

```diff
@@ -97,4 +104,54 @@ class GraphQlApiSpec extends ExportSpec {
     }
     exception.getMessage should equal(s"No data returned from the update export call for consignment $consignmentId ")
   }
+
+  "the getConsignmentMetadata method" should "return the correct value" in {
+    val fixedDate = ZonedDateTime.now()
+    val userId = UUID.randomUUID()
+    val series = Series(Some("series-code"))
+    val transferringBody = TransferringBody(Some("tb-code"))
+    val consignmentClient = mock[GraphQLClient[gce.Data, gce.Variables]]
+    val filesClient = mock[GraphQLClient[gf.Data, gf.Variables]]
+    val updateExportClient = mock[GraphQLClient[uel.Data, uel.Variables]]
+    val getOriginalPathClient = mock[GraphQLClient[gop.Data, gop.Variables]]
+    val keycloak = mock[KeycloakUtils]
+    val api = new GraphQlApi(keycloak, consignmentClient, filesClient, updateExportClient, getOriginalPathClient)
+    val config = Configuration(S3("", "", ""), Api(""), Auth("authUrl", "clientId", "clientSecret", "realm"), EFS(""))
+    val consignmentId = UUID.randomUUID()
+    val consignment = GetConsignmentExport.getConsignmentForExport.GetConsignment(
+      userId, Some(fixedDate), Some(fixedDate), Some(fixedDate), Some(series), Some(transferringBody)
+    )
+
+    doAnswer(() => Future(new BearerAccessToken("token"))).when(keycloak).serviceAccountToken[Identity](any[String], any[String])(any[SttpBackend[Identity, Nothing, NothingT]], any[ClassTag[Identity[_]]])
+    val data = new GraphQlResponse[gce.Data](gce.Data(Some(consignment)).some, List())
+    doAnswer(() => Future(data)).when(consignmentClient).getResult[Identity](any[BearerAccessToken], any[Document], any[Option[gce.Variables]])(any[SttpBackend[Identity, Nothing, NothingT]], any[ClassTag[Identity[_]]])
```

Comment:
There are lots arguments like `any[Foo]` here, which means that we're not really testing that the getConsignmentMetadata is responding to the right input. Is it hard to test it with the real values?

Reply:
All the values will need to be objects, cannot mix matchers with raw objects. Therefore either have to create all the objects, and some are very complex, eg AST document, or use mocks.